starbound/btreedb5.py

```diff
@@ -66,6 +66,41 @@ def get(self, key):
         # None of the keys in the leaf node matched.
         raise KeyError(binascii.hexlify(key))
 
+    def get_all_keys(self, start=None):
+        """
+        Returns a list of all valid keys starting at the given `start`
+        offset.  If `start` is `None`, we will start from the root of
+        the tree.
+        """
+        keys = []
+        s = self.stream
+        if not start:
+            start = HEADER_SIZE + self.block_size * self.root_block
+        s.seek(start)
+        block_type = s.read(2)
+        if block_type == LEAF:
+            reader = LeafReader(self)
+            num_keys = struct.unpack('>i', reader.read(4))[0]
+            for i in range(num_keys):
+                cur_key = reader.read(self.key_size)
+                keys.append(cur_key)
+                length = sbon.read_varint(reader)
+                reader.seek(length, 1)
```

Comment:
This `read_varint` and `seek` doesn't seem to be used so could be avoided, right?

Reply:
(aha, probably should've responded in here to begin with).  Anyway, the `read_varint` and `seek` look to be necessary whenever there's more than one key in a node.  Don't think we can do without those...